db/migrate/20220818074150_fix_invalid_journals.rb

```diff
@@ -0,0 +1,28 @@
+class FixInvalidJournals < ActiveRecord::Migration[7.0]
+  def up
+    get_broken_journals.each do |journable_type, relation|
+      next unless relation.any?
+
+      puts "Cleaning up broken journals on #{journable_type}"
+      relation.destroy_all
+    end
+  end
+
+  def down
+    # nothing to do
+  end
+
+  def get_broken_journals
+    Journal
+      .pluck('DISTINCT(journable_type)')
+      .to_h do |journable_type|
+      journal_class = journable_type.constantize.journal_class
```

Comment:
Locally on my instance I got a `undefined method 'constantize' for nil:NilClass` error.

Indeed I have one `Journal` with `journable_type: nil`

```
> Journal.where(journable_type:nil)
  Journal Load (0.9ms)  SELECT "journals".* FROM "journals" WHERE "journals"."journable_type" IS NULL
=> [#<Journal:0x00007fededbf8e88
  id: 835,
  journable_type: nil,
  journable_id: nil,
  user_id: 0,
  notes: nil,
  created_at: Thu, 25 Aug 2022 08:00:00.000000000 UTC +00:00,
  version: 0,
  activity_type: nil,
  updated_at: Thu, 25 Aug 2022 08:00:00.000000000 UTC +00:00,
  data_type: nil,
  data_id: nil>]
```

I do not know if that's common or not. It may just be me having created this data for some reason and forgot about it.

Reply:
I haven't seen this so far but just added a `.compact` in between to ensure it doesn't happen